src/main/java/org/dita/dost/module/MergeDitavalModule.java

```diff
@@ -211,55 +207,29 @@ private void updateBaseDirectory() {
             baseInputDir = baseInputDir.resolve("..");
         }
     }
-
+    
     /**
-     * Write result files.
-     * 
-     * @throws DITAOTException
-     *             if writing result files failed
+     * Set external property with the name of the merged file.
      */
-    private void outputResult() throws DITAOTException {
-        try {
-            tempFileNameScheme = (TempFileNameScheme) getClass().forName(job.getProperty("temp-file-name-scheme"))
-                    .newInstance();
-        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
-            throw new RuntimeException(e);
-        }
-        tempFileNameScheme.setBaseDir(baseInputDir);
-
-        // assume empty Job
-        final URI rootTemp = tempFileNameScheme.generateTempFileName(rootFile);
-        final File relativeRootFile = toFile(rootTemp);
-
-        if (baseInputDir.getScheme().equals("file")) {
-            job.setProperty(INPUT_DIR, new File(baseInputDir).getAbsolutePath());
-        }
-        job.setProperty(INPUT_DIR_URI, baseInputDir.toString());
-        job.setProperty(INPUT_DITAMAP, relativeRootFile.toString());
-        job.setProperty(INPUT_DITAMAP_URI, rootTemp.toString());
-
-        writeMergedDitaval();
-
+    private void setMergedProperty() {
+        job.setProperty("dita.input.valfile", new File(job.tempDir, FILE_NAME_MERGED_DITAVAL).toString());
     }
 
     private void writeMergedDitaval() throws DITAOTException {
         XMLStreamWriter export = null;
-        try (OutputStream exportStream = new FileOutputStream(new File(job.tempDir, mergedDitavalFilename))) {
+        try (OutputStream exportStream = new FileOutputStream(new File(job.tempDir, FILE_NAME_MERGED_DITAVAL))) {
             export = XMLOutputFactory.newInstance().createXMLStreamWriter(exportStream, "UTF-8");
             export.writeStartDocument();
             export.writeStartElement("val");
             export.writeNamespace(DITA_OT_NS_PREFIX, DITA_OT_NS);
-            if (!ditavalFiles.isEmpty()) {
-                final Iterator<File> dvi = ditavalFiles.iterator();
-                while (dvi.hasNext()) {
-                    final File curDitaVal = dvi.next();
-                    final DocumentBuilder ditavalbuilder = XMLUtils.getDocumentBuilder();
-                    final Document doc = ditavalbuilder.parse(curDitaVal);
-                    final Element ditavalRoot = doc.getDocumentElement();
-                    final NodeList rootChildren = ditavalRoot.getChildNodes();
-                    logger.debug("Writing conditions from ditaval: " + curDitaVal);
-                    writeConditions(export, rootChildren, curDitaVal.toURI().resolve("."));
-                }
+            for (int i = 0; i < ditavalFiles.size(); i++) {
```

Comment:
Since `ditavalFiles` is a `List`, this should not use a [for statement](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html), but rather [for-each loop](https://docs.oracle.com/javase/1.5.0/docs/guide/language/foreach.html).